var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -763,3 +836,113 @@ def check(self):
         with spack.util.environment.set_env(CP2K_DATA_DIR=data_dir, PWD=self.build_directory):
             with working_dir(self.build_directory):
                 make("test", *self.build_targets)
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    """Use the new cmake build system to build cp2k. It is the default when
+    building the master branch of cp2k."""
+
+    def cmake_args(self):
+        spec = self.spec
+        args = []
+
+        gpu_map = {
+            "35": "K40",
+            "37": "K80",
+            "60": "P100",
+            "70": "V100",
+            "80": "A100",
+            "gfx906": "Mi50",
+            "gfx908": "Mi100",
+            "gfx90a": "Mi250",
+            "gfx90a:xnack-": "Mi250",
+            "gfx90a:xnack+": "Mi250",
+        }
```

Comment:
The same `gpu_map` is used above, is it worth defining it in one place (globally)?